src/lib/utils/codec_base.h

```diff
@@ -35,8 +36,8 @@ size_t base_encode(
    Base&& base, char output[], const uint8_t input[], size_t input_length, size_t& input_consumed, bool final_inputs) {
    input_consumed = 0;
 
-   const size_t encoding_bytes_in = base.encoding_bytes_in();
-   const size_t encoding_bytes_out = base.encoding_bytes_out();
+   constexpr size_t encoding_bytes_in = std::remove_reference_t<Base>::encoding_bytes_in();
```

Comment:
Could you here (and also below) add a comment along the lines of 

```
// TODO(Botan4) Check if we can use just base. or Base:: here instead
```

Reply:
Done.